src/shader_recompiler/ir/ir_emitter.cpp

```diff
@@ -145,34 +145,64 @@ void IREmitter::SetThreadBitScalarReg(IR::ScalarReg reg, const U1& value) {
 
 template <>
 U32 IREmitter::GetScalarReg(IR::ScalarReg reg) {
-    return Inst<U32>(Opcode::GetScalarRegister, reg);
+    return Inst<U32>(Opcode::GetScalarRegister, reg, Imm32(32));
 }
 
 template <>
 F32 IREmitter::GetScalarReg(IR::ScalarReg reg) {
     return BitCast<F32>(GetScalarReg<U32>(reg));
 }
 
+template <>
+U64 IREmitter::GetScalarReg(IR::ScalarReg reg) {
+    return Inst<U64>(Opcode::GetScalarRegister, reg, Imm32(64));
```

Comment:
This is more complicated than necessary I think. You can avoid modifying the IR instruction by loading 2 scalar/vector registers and combining them with PackUint2x32. So this instruction is kept clean

Reply:
Also IREmitter should not handle this at all I think. Have Translator do this in 64-bit version of GetSrc

Reply:
This also depends on how guest shader uses 64-bit arithmetic. For single instances in some instructions I feel this will be a simpler way to achieve the result in the current system. Otherwise if a larger section of the shader performs shader arithmetic it might be worth investigating other solutions. Probably a constant propagation pass that detects this Pack -> Unpack patterns and preserves the original U64 value